### ISSUE TITLE
New Rule: no-floating-promise

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#43]
## Research Evidence

From empirical studies of LLM-generated code bugs:
- Missing/incorrect code blocks account for 40%+ of syntactic errors
- LLMs often create promises that are never awaited or returned

## Description

Detect Promises that are created but never awaited, returned, or explicitly handled. This is a common LLM mistake where async functions are called without awaiting the result.

### Bad
```ts
async function saveData() { ... }

// LLM forgot to await
saveData(); // Promise floats away

// Stored but never used
const result = fetchData();
console.log('done'); // forgot to await result
```

### Good
```ts
await saveData();
const result = await fetchData();
void saveData(); // Explicit fire-and-forget
```

## Notes
- TypeScript has `noFloatingPromises` but requires strict configuration
- This would be an ESLint-level guard with LLM-friendly error messages
- Should allow `void` prefix for intentional fire-and-forget

## Priority
High - very common LLM mistake

**Comment by pertrai1** [comment]
## Analysis: Not ready to implement

`@typescript-eslint/no-floating-promises` already exists in `recommendedTypeChecked` and does exactly what this issue describes — flags promises that aren't awaited, returned, or voided.

The key question this issue doesn't address: **can floating promises be reliably detected without type information?**

Without `parserServices` (type-aware linting), the rule can only detect unawaited calls on functions explicitly declared `async`. It would miss:
- `fetch()`, `Promise.resolve()`, and any non-async function that returns a Promise
- Promises stored in variables and never consumed

The false negative rate would be very high, and there's false positive risk (flagging non-promise function calls that look async).

**Before implementing, we need to determine:**
1. What subset of floating promises can be detected without type info?
2. Is that subset large enough to justify a rule when `@typescript-eslint/no-floating-promises` already covers it comprehensively?
3. Would the LLM-specific teaching message alone justify shipping a narrower version?

Relabeled to `needs-analysis`.
